flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamPhysicalMultiJoin.java

```diff
@@ -237,6 +252,104 @@ public List<JoinRelType> getJoinTypes() {
+            commonJoinKey.add("noCommonJoinKey");
+        }
+
+        return String.join(", ", commonJoinKey);
+    }
+
+    /**
+     * Formats a RexNode expression with field names for better readability in explain output.
+     *
+     * @param expression the expression to format
+     * @param pw the RelWriter for determining format preferences
+     * @return formatted expression string with field names
+     */
+    private String formatExpressionWithFieldNames(final RexNode expression, final RelWriter pw) {
+        if (expression == null) {
+            return "";
+        }
+
+        return getExpressionString(
+                expression,
+                JavaScalaConversionUtil.toScala(getRowType().getFieldNames()).toList(),
+                JavaScalaConversionUtil.toScala(Optional.empty()),
+                RelExplainUtil.preferExpressionFormat(pw),
+                RelExplainUtil.preferExpressionDetail(pw));
+    }
+
+    /**
+     * Formats join conditions with field names for better readability in explain output.
+     *
+     * @param pw the RelWriter for determining format preferences
+     * @return formatted join conditions string with field names
+     */
+    private String formatJoinConditionsWithFieldNames(final RelWriter pw) {
+        if (joinConditions.isEmpty()) {
+            return "";
+        }
+
+        final List<String> formattedConditions = new ArrayList<>();
+        for (final RexNode condition : joinConditions) {
+            if (condition != null) {
+                formattedConditions.add(formatExpressionWithFieldNames(condition, pw));
+            }
+        }
+
+        return String.join(", ", formattedConditions);
+    }
+
+    private String formatInputUniqueKeysWithFieldNames() {
+        final List<String> inputUniqueKeyStrings = new ArrayList<>();
+        for (int i = 0; i < inputs.size(); i++) {
+            final RelNode input = inputs.get(i);
+            final List<String> fieldNames = input.getRowType().getFieldNames();
+            final Set<ImmutableBitSet> uniqueKeys = getUniqueKeys(input);
+
+            if (uniqueKeys != null && !uniqueKeys.isEmpty()) {
+                final List<String> uniqueKeyStrings = new ArrayList<>();
+                for (final ImmutableBitSet uniqueKey : uniqueKeys) {
+                    final List<String> keyFieldNames = new ArrayList<>();
+                    for (final int index : uniqueKey.toArray()) {
+                        if (index < fieldNames.size()) {
+                            keyFieldNames.add(fieldNames.get(index));
+                        }
+                    }
+                    if (!keyFieldNames.isEmpty()) {
+                        uniqueKeyStrings.add("(" + String.join(", ", keyFieldNames) + ")");
+                    }
+                }
+                if (!uniqueKeyStrings.isEmpty()) {
+                    inputUniqueKeyStrings.add(String.join(", ", uniqueKeyStrings));
+                }
```

Comment:
```suggestion
                inputUniqueKeyStrings.add(String.join(", ", uniqueKeyStrings));
```
nit: it looks like we don't need this condition since if it is empty it will add an empty string